src/TrackableTrait.php

```diff
@@ -9,14 +9,53 @@
  */
 trait TrackableTrait {
 
+    /**
+     * Check this property to see if TrackableTrait is present
+     * in the object
+     *
+     * @var string
+     */
+    public $_trackableTrait = true;
+
+    /**
+     * Link to object into which we added this object
+     *
+     * @var AbstractObject
+     */
+    public $owner;
+
+    /**
+     * Unique object name
+     *
+     * @var string
+     */
     public $name;
+
+    /**
+     * Name of the object in owner's element array
+     *
+     * @var string
+     */
     public $short_name;
 
+    /**
+     * If name of the object is ommitted then it's natuarlly to name them
+     * after the class. You can specify a different naming pattern though
+     */
     function getDesiredName(){
-        return get_class($this);
+        return str_replace('\\', '_', strtolower(get_class($this)));
     }
 
+    /**
+     * Removes object from parent, so that PHP's Garbage Collector can
+     * dispose of it.
+     */
     function destroy(){
-        unset($this->owner->elements[$this->short_name]);
+        if (
+            isset($this->owner) &&
+            $this->owner->_containerTrait
```

Comment:
this will throw warning if owner doesn't have _containerTrait property (or contant) at all. Should use isset() here. Isset "_Returns TRUE if var exists and has value other than NULL, FALSE otherwise._"